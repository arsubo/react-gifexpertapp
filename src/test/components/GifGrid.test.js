import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); //finge el llamado a este componente

describe('Evaluación del componente <GifGrid />', () => {
    const category = 'One Punch';


    test('Evaluar que el componente carga correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe de mostrar items cuando se cargan imágenes con useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }]

        //enviamos data falsa
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);

        //evaluamos si cargó correctamente
        expect(wrapper).toMatchSnapshot();
        //Si cargó los componentes no debería existir un parrafo
        expect(wrapper.find('p').exists()).toBe(false);
        //evaluamos si creó el objeto GifGridItem
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);


    })

})
