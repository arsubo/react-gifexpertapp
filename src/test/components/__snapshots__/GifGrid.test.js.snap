// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluación del componente <GifGrid /> Debe de mostrar items cuando se cargan imágenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Evaluación del componente <GifGrid /> Evaluar que el componente carga correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
